File: ASM2_API/ASM-Java5/src/main/java/com/dto/user/UserUpdateRequest.java
package com.dto.user;

import java.time.LocalDate;

public record UserUpdateRequest(
        Long userId,
        String email,
        String username,
        String firstName,
        String lastName,
        LocalDate birthday,
        String password,
        String phoneNumber,
        Integer roleId,
        Boolean isActive
) {
}

File: ASM2_API/ASM-Java5/src/main/kotlin/com/dto/user/UserUpdateRequest.kt
package com.dto.user

import java.time.LocalDate

data class UserUpdateRequest(
    val userId: Long,  // Required to identify which user to update
    val email: String? = null,
    val username: String? = null,
    val firstName: String? = null,
    val lastName: String? = null,
    val birthday: LocalDate? = null,
    val password: String? = null,  // Optional password change
    val phoneNumber: String? = null,
    val roleId: Int? = null,
    val isActive: Boolean? = null
)
